rabix-engine/src/main/java/org/rabix/engine/service/impl/JobServiceImpl.java

```diff
@@ -381,13 +381,19 @@ private boolean isFinished(JobStatus jobStatus) {
   @Override
   public void handleJobContainerReady(Job containerJob) {
     logger.info("Container job {} rootId: {} id ready.", containerJob.getName(), containerJob.getRootId());
-    if (deleteIntermediaryFiles) {
-      intermediaryFilesService.handleContainerReady(containerJob, keepInputFiles);
-    }
     try {
       engineStatusCallback.onJobContainerReady(containerJob);
     } catch (EngineStatusCallbackException e) {
       logger.error("Engine status callback failed", e);
+    }    
+    if (deleteIntermediaryFiles) {
+      Runnable r = new Runnable() {
+        public void run() {
+          intermediaryFilesService.handleContainerReady(containerJob, keepInputFiles);
```

Comment:
This shouldn't be done in an asynchronous way. There's a potential problem with deleting some of the data after handling an external event. This needs to be processed from the event processor thread.